### PR TITLE
Added new properties to the Subscriptions table

### PR DESCRIPTION
**Pull request by alexdrenea**
- OfferId (Name) and Term details (Name, StartDate, EndDate)
- Updated subscriptions page both admin and customer to show Offer Name
- Updated subscriptions details page to show term details

Note!
For customers upgrading from an older version of accelerator, they should use the upgrade.ps1 script to upgrade the code and database, then run a "Fetch all Subscriptions" to populate the new fields.

Addresses #548 

Subscriptions page:
image

Subscription Details Page:
image

### CODE REVIEW

File: src/DataAccess/Migrations/20230912052848_SubscriptionDetails.cs
Comment (santhoshb-msft) [inline src/DataAccess/Migrations/20230912052848_SubscriptionDetails.cs:1]: @alexd would it be possible to name these files ending with _v714 to denote the future version please?

Comment (alexdrenea) [reply src/DataAccess/Migrations/20230912052848_SubscriptionDetails.cs:1]: done, renamed to _v740

**Review by santhoshb-msft — CHANGES_REQUESTED** [review]
Thank you for making the migration and Ui changes. 
@alexdrenea could you please merge your branch from main latest.

**Review by santhoshb-msft — APPROVED** [review]
verified in both new and upgrade scenarios.

File: src/DataAccess/Migrations/20230912052848_SubscriptionDetails.cs
Comment (santhoshb-msft) [reply src/DataAccess/Migrations/20230912052848_SubscriptionDetails.cs:1]: thank you @alexdrenea

**Pull request merged**